core/src/main/scala/org/apache/spark/rdd/RDD.scala

```diff
@@ -1539,6 +1539,9 @@ abstract class RDD[T: ClassTag](
     // NOTE: we use a global lock here due to complexities downstream with ensuring
     // children RDD partitions point to the correct parent partitions. In the future
     // we should revisit this consideration.
+    if (doCheckpointCalled) {
+      logWarning(s"Because job has been executed on RDD ${id}, checkpoint won't work")
```

Comment:
We have to decide if we simply log a message and ignore it (if it is a design choice) or if we need to fix this.
From git, @mateiz and @zsxwing were the last people to work on it this - would be great to hear from them : this might be a variant of SPARK-6847 which @zsxwing fixed.